src/webui/www/private/scripts/client.js

```diff
@@ -1387,11 +1387,11 @@ function registerMagnetHandler() {
         return;
     }
 
-    const hashParams = getHashParamsFromUrl();
-    hashParams.download = '';
-
-    const templateHashString = Object.toQueryString(hashParams).replace('download=', 'download=%s');
+    const hashString = location.hash ? location.hash.replace(/^#/, '') : '';
+    const hashParams = new URLSearchParams(hashString);
```

Comment:
How is it intended to behave in case when `hashString` is empty at this place?

Reply:
I only replaced the intermediate API, the rest is still the same.